index.bs

```diff
@@ -52,10 +66,169 @@ dictionary StorageBucketOptions {
 </xmp>
 
 <h3 id="storage-bucket-open">Creating a bucket</h3>
+
+<div algorithm>
+
+The <dfn method for="StorageBucketManager">open(|name|, |options|)</dfn> method steps are:
+
+1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+
+1. Let |storageKey| be the result of running [=obtain a storage key=] given |environment|.
+
+1. If |storageKey| is failure, then [=exception/throw=] a "{{SecurityError}}" {{DOMException}} and abort these steps.
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. If the result of [=validate a bucket name=] with |name| is failure, then [=/reject=] |p| with an {{InvalidCharacterError}} and abort these steps.
+
+    1. Let |r| be the result of running [=open a bucket=] with |storageKey|, |name|, and |options|.
+
+    1. If |r| is failure, then [=reject=] |p| with a {{TypeError}} and abort these steps.
+
+    1. [=/Resolve=] |p| with |r|.
+
+1. Return |p|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>open a bucket</dfn> for a |storageKey| given a bucket |name| and optional |options|, run the following steps:
+
+1. Let |expires| be |options|["{{StorageBucketOptions/expires}}"] if it exists, otherwise undefined.
+
+1. If |expires| is not undefined, and is less than or equal to now, then return failure.
```

Comment:
Do you need to handle some garbage string being passed here?  Should we throw if expires is not a number or undefined?  Also, we should probably be explicit about which clock we are using.  Its not clear to me if this is comparing milliseconds or seconds since epoch or what.

Reply:
Updated to use `[=parse a duration string=]` like in spec example https://w3c.github.io/hr-time/#sec-tools-examples